app/models/concerns/reminders/states.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Reminders::States
+  extend ActiveSupport::Concern
+
+  STATUSES = %w(pending scheduled canceled done)
+
+  included do
+    before_validation :set_default_state_class_type
+  end
+
+  def state
+    "Reminders::States::#{state_class_type.capitalize}".constantize.new
+  end
+
+  def scheduled
+    self.update state_class_type: 'scheduled'
```

Comment:
```suggestion
    update! state_class_type: 'scheduled'
```
Por si falla mejor si vamos con ese, de la otra forma puede "parecer" que hizo el cambio y en realidad por algún otro tema puede que no.

Reply:
totalmente de acuerdo